File: app/src/main/java/sgtmelon/scriptum/office/intf/ItemListener.java
package sgtmelon.scriptum.office.intf;

import android.view.View;

import androidx.annotation.NonNull;

/**
 * Интерфейс для обработки нажатий на элементы
 * На Java, чтобы использовать lambda в Kotlin
 */
public interface ItemListener {

    interface Click {
        void onItemClick(@NonNull View view, int p);
    }

    interface LongClick {
        void onItemLongClick(@NonNull View view, int p);
    }

    interface Drag {
        void setDrag(boolean drag);
    }

}

File: app/src/main/java/sgtmelon/scriptum/listener/ItemListener.kt
package sgtmelon.scriptum.listener

import android.view.View

/**
 * Interface for different touches on view
 */
interface ItemListener {

    interface Click {
        fun onItemClick(view: View, p: Int)
    }

    interface LongClick {
        fun onItemLongClick(view: View, p: Int)
    }

    interface Drag {
        fun setDrag(drag: Boolean)
    }

}